File: square/src/main/java/com/raxdenstudios/square/manager/ApplicationMultiDexInterceptorManager.java
package com.raxdenstudios.square.manager;

import android.content.res.Configuration;

import com.raxdenstudios.square.interceptor.ApplicationInterceptor;
import com.raxdenstudios.square.lifecycle.ApplicationLifecycle;

import androidx.multidex.MultiDexApplication;

/**
 * Created by Ángel Gómez on 18/12/2016.
 */

public class ApplicationMultiDexInterceptorManager
        extends InterceptorManager<MultiDexApplication, ApplicationInterceptor>
        implements ApplicationLifecycle {

    public ApplicationMultiDexInterceptorManager(MultiDexApplication application) {
        super(application);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onConfigurationChanged(newConfig);
        }
    }

    @Override
    public void onCreate() {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onCreate();
        }
    }

    @Override
    public void onLowMemory() {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onLowMemory();
        }
    }

    @Override
    public void onTrimMemory(int level) {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onTrimMemory(level);
        }
    }

}

File: square/src/main/java/com/raxdenstudios/square/manager/ApplicationMultiDexInterceptorManager.kt
package com.raxdenstudios.square.manager

import android.content.res.Configuration
import android.support.multidex.MultiDexApplication

import com.raxdenstudios.square.interceptor.ApplicationInterceptor
import com.raxdenstudios.square.lifecycle.ApplicationLifecycle

/**
 * Created by Ángel Gómez on 18/12/2016.
 */
class ApplicationMultiDexInterceptorManager(application: MultiDexApplication) : InterceptorManager<MultiDexApplication, ApplicationInterceptor<*>>(application), ApplicationLifecycle {

    override fun onConfigurationChanged(newConfig: Configuration?) {
        interceptors.forEach { it.onConfigurationChanged(newConfig) }
    }

    override fun onCreate() {
        interceptors.forEach { it.onCreate() }
    }

    override fun onLowMemory() {
        interceptors.forEach { it.onLowMemory() }
    }

    override fun onTrimMemory(level: Int) {
        interceptors.forEach { it.onTrimMemory(level) }
    }

}
